src/manager.rs

```diff
@@ -186,8 +186,16 @@ impl<'a> KeyHolderWrite<'a> {
         if paths.is_empty() {
             // updating the root require special treatment
             let root = self.get_value().unwrap().unwrap();
-            let res = (op_fun)(root.take())?;
-            self.set_root(res)?;
+            let prev_val = root.take();
+            let rollback_val = prev_val.clone();
```

Comment:
So you do use clone?

Reply:
This is outdated.
Checking where is commit ea44cf5 ?

Reply:
How come its outdated and I see it in this PR?

Reply:
This was a preliminary commit which was also addressing `serde_json`.
Now we currently putting `serde_json` aside.
I suggest to keep it anyway (no need to rollback IMHO)